File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        max_area = 0
        for i in range(len(heights)):
            l = i
            r = i + 1
            height = heights[i]
            while l >= 0 and heights[l] >= height: 
                #calculate the area
                l-=1
            while r < len(heights) and heights[r] >= height:
                r+=1
            
            r-=1
            l+=1
            width = (r - l + 1)
            max_area = max(max_area, self.calculateArea(height, width))
        return max_area
            
    
    def calculateArea(self, height: int, width: int):
        return height * width

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int maximun = 0;
        int l = 0;
        int r = heights.length - 1;
        // to calculate the height lets start why iterating
        while(l < r ) {
            int distance = r - l;
            int area = distance * Math.min(heights[r], heights[l]);
            maximun = Math.max(area,maximun);
            // lets move the lower bar;
            if(heights[l] < heights[r]) {
                l++;
            }else {
                r--;
            }
        }
        return maximun;
    }
}
